UnitySDK/Assets/ML-Agents/Scripts/DecisionAutoRequester.cs

```diff
@@ -0,0 +1,33 @@
+﻿using UnityEngine;
+
+namespace MLAgents{
+    [RequireComponent(typeof(Agent))]
+    public class DecisionAutoRequester : MonoBehaviour {
+
+        public int DecisionPeriod;
+        public bool RepeatAction;
+        private Agent m_Agent;
+
+        public void OnEnable()
+        {
+            DecisionPeriod = Mathf.Max(DecisionPeriod, 1);
+            m_Agent = gameObject.GetComponent<Agent>();
+            GameObject.FindObjectOfType<Academy>().AgentSetStatus += SetStatus;
+        }
+
+        void OnDisable()
+        {
+            GameObject.FindObjectOfType<Academy>().AgentSetStatus -= SetStatus;
+        }
+
+        void SetStatus(bool academyMaxStep, bool academyDone, int academyStepCounter)
+        {
+            if(academyStepCounter % DecisionPeriod == 0){
```

Comment:
Also, right now we are using the Academy SetStatus. Maybe this is a bad idea. The reason behind using SetStatus is so that all of the agents are in sync (regardless of when they are instantiated) but maybe it would be better to remove this all together and add this "offset" you talk about. @chriselion @surfnerd Should we discuss this change further ?